components/list/list-item-drag-mixin.js

```diff
@@ -0,0 +1,498 @@
+		};
+	}
+
+	static get styles() {
+		const styles = [ css`
+			:host {
+				display: block;
+				position: relative;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			.d2l-list-item-drag-bottom-marker,
+			.d2l-list-item-drag-top-marker {
+				position: absolute;
+				width: 100%;
+				z-index: 1;
+			}
+			.d2l-list-item-drag-bottom-marker {
+				bottom: -6px;
+			}
+			.d2l-list-item-drag-top-marker {
+				top: -6px;
+			}
+			.d2l-list-item-drag-area {
+				cursor: move;
+				height: 100%;
+			}
+			.d2l-list-item-drag-drop-grid {
+				display: grid;
+				grid-template-columns: 100%;
+				grid-template-rows: 1rem 1fr 1fr 1rem;
+				height: 100%;
+				position: absolute;
+				top: 0;
+				width: 100%;
+				z-index: 100;
+			}
+			d2l-list-item-drag-handle {
+				opacity: 0;
+			}
+			d2l-list-item-drag-handle.d2l-hovering,
+			d2l-list-item-drag-handle.d2l-focusing {
+				opacity: 1;
+			}
+		` ];
+
+		super.styles && styles.unshift(super.styles);
+		return styles;
+	}
+
+	constructor() {
+		super();
+		this._itemDragId = getUniqueId();
+		this.dragging = false;
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		if (!this.key) {
+			this.draggable = false;
+		}
+	}
+
+	firstUpdated(changedProperties) {
+		this.addEventListener('mouseenter', this._onMouseEnter.bind(this));
+		this.addEventListener('mouseleave', this._onMouseLeave.bind(this));
+		this.addEventListener('dragenter', this._onHostDragEnter);
+		super.firstUpdated(changedProperties);
+	}
+
+	_annoucePositionChange(dragTargetKey, dropTargetKey, dropLocation) {
+		this.dispatchEvent(new CustomEvent('d2l-list-item-position-change', {
+			detail: new NewPositionEventDetails({dragTargetKey, dropTargetKey, dropLocation}),
+			bubbles: true
+		}));
+	}
+
+	_onDragAreaClick(e) {
+		this.shadowRoot.querySelector(`#${this._itemDragId}`).activateKeyboardMode();
+		e.preventDefault();
+	}
+
+	_onDragEnd(e) {
+		const dropSpot = getDragState();
+		this.dragging = false;
+		if (dropSpot.shouldDrop(e.timeStamp)) {
+			this._annoucePositionChange(dropSpot.dragTargetKey, dropSpot.dropTargetKey, dropSpot.dropLocation);
+		}
+		clearDragState();
+	}
+
+	_onDragHandleActions(e) {
+		switch (e.detail.action) {
+			case dragActions.active:
+				this._keyboardActive = true;
+				break;
+			case dragActions.cancel:
+			case dragActions.save:
+				this._keyboardActive = false;
+				break;
+			case dragActions.up:
+				this._annoucePositionChange(this.key, null, dropLocation.shiftUp);
+				break;
+			case dragActions.down:
+				this._annoucePositionChange(this.key, null, dropLocation.shiftDown);
+				break;
+			case dragActions.first:
+				this._annoucePositionChange(this.key, null, dropLocation.first);
+				break;
+			case dragActions.last:
+				this._annoucePositionChange(this.key, null, dropLocation.last);
+				break;
+			default:
+				break;
+		}
+	}
+
+	_onDragOver(e) {
+		if (!this.key) return;
+		const dropSpots = getDragState();
+		dropSpots.updateTime(e.timeStamp);
```

Comment:
I think these two lines should be:
```
const dragState = getDragState();
dragState.updateTime(e.timeStamp);
```
... just missed during the rename I think.